be/src/vec/functions/function_string.h

```diff
@@ -1344,6 +1345,167 @@ class FunctionSplitPart : public IFunction {
     }
 };
 
+class FunctionSplitByChar : public IFunction {
+public:
+    static constexpr auto name = "split_by_char";
+
+    static FunctionPtr create() { return std::make_shared<FunctionSplitByChar>(); }
+    using NullMapType = PaddedPODArray<UInt8>;
+
+    String get_name() const override { return name; }
+
+    bool is_variadic() const override { return false; }
+
+    size_t get_number_of_arguments() const override { return 2; }
+
+    DataTypePtr get_return_type_impl(const DataTypes& arguments) const override {
+        DCHECK(is_string(arguments[0]))
+                << "first argument for function: " << name << " should be string"
+                << " and arguments[0] is " << arguments[0]->get_name();
+        DCHECK(is_string(arguments[1]))
+                << "second argument for function: " << name << " should be char"
+                << " and arguments[1] is " << arguments[1]->get_name();
+        return std::make_shared<DataTypeArray>(make_nullable(arguments[0]));
+    }
+
+    Status execute_impl(FunctionContext* context, Block& block, const ColumnNumbers& arguments,
+                        size_t result, size_t input_rows_count) override {
+        DCHECK_EQ(arguments.size(), 2);
+
+        ColumnPtr src_column = make_nullable(
+                block.get_by_position(arguments[0]).column->convert_to_full_column_if_const(),
+                false);
+        ColumnPtr delimiter_column =
+                block.get_by_position(arguments[1]).column->convert_to_full_column_if_const();
+        for (size_t i = 0; i < input_rows_count; ++i) {
+            string delimiter = delimiter_column->get_data_at(i).to_string();
+            if (delimiter.size() > 1) {
+                return Status::RuntimeError(
+                        fmt::format("only supported one or zero character delimiter for function "
+                                    "{}(but get {})",
+                                    get_name(), delimiter.size()));
+            }
+        }
+
+        DataTypePtr src_column_type = make_nullable(block.get_by_position(arguments[0]).type);
+        auto dest_column_ptr = ColumnArray::create(src_column_type->create_column(),
+                                                   ColumnArray::ColumnOffsets::create());
+        IColumn* dest_nested_column = &dest_column_ptr->get_data();
+        auto& dest_offsets = dest_column_ptr->get_offsets();
+        DCHECK(dest_nested_column != nullptr);
+        dest_nested_column->reserve(0);
+        dest_offsets.reserve(0);
+
+        const NullMapType* src_null_map = nullptr;
+        if (src_column->is_nullable()) {
```

Comment:
do not need check the src_column `is_nullable`, the function is `depend on argument`. the args is always not nullable